### DISCUSSION TITLE
ESP32 purchase question

### DISCUSSION THREAD

**Discussion by brmeadows2841** [forkineye/ESPixelStick#486] [Q&A]
With features being added to the ESP32 platform, I have a question about which to purchase in the near future.

There are ESP32s with PSRAM, Ethernet, SD card slot, external antenna, etc. Most of the ESP32 with SD card slot have a camera, which eats up GPIO pins. 
Anyone know if there is an ESP32 with all the above features, except external antenna?
I looking for an ESP32 Wrover with ethernet and SD Card.

**Comment by MartinMueller2003** [comment]
My favorite card is the LoLin D32 Pro. Connecting Ethernet to it is a PITA so I am experimenting with the WT32-ETH01. Software is not stable for it yet but it will be soon. I do not like the ESP32-CAM variants. The SD Card implementation is very unstable.

**Reply by vrtx-voiod** [reply]
I am quite new to the whole ESPixelStick game and quite interested in using Wemos D1 Mini ESP32 modules. However I still cannot wrap my head around how to deal with the GPIO layout: Can I just attach an ESP32 on the ESPixelStick V3 pin board, or do I have to do any further modifications to make that setup work?

**Comment by MartinMueller2003** [comment]
it depends on the pin layout. I use a quinled ESP32 module plugged into the V3 base card. I had to bend the EN pin so it did not get inserted in the socket. Adjusted the GPIO pins for SD and pixel output and away it went. I also got the version with the external antenna to get a longer reach on the wifi
